entrypoint.sh

```diff
@@ -41,43 +57,55 @@ chown nginx:nginx /var/cache/nginx
 mkdir -p /var/tmp/nginx
 chown nginx:nginx /var/tmp/nginx
 
+#create vhost directory
+mkdir -p /etc/nginx/vhosts/
+
+# Copy nginx main conf file.
+cp templates/nginx.conf /etc/nginx/nginx.conf
+
 # Process templates
-for t in /templates/*
+upstreamId=0
+for t in "${DOMAINSARRAY[@]}"
 do
-  dest="/etc/nginx/$(basename $t)"
-  echo "Rendering template of $dest"
-  sed -e "s/\${DOMAIN}/${DOMAIN}/g" \
-      -e "s/\${UPSTREAM}/${UPSTREAM}/" \
-      $t > $dest
+  dest="/etc/nginx/vhosts/$(basename "${t}").conf"
+  echo "Rendering template of $t in $dest"
+  sed -e "s/\${DOMAIN}/${t}/g" \
+      -e "s/\${UPSTREAM}/${UPSTREAMARRAY[upstreamId]}/" \
+      /templates/vhost.sample.conf > "$dest"
+  upstreamId=$((upstreamId+1))
 done
 
+
+for d in "${DOMAINSARRAY[@]}"
+do
+
 # Initial certificate request, but skip if cached
-if [ ! -f /etc/letsencrypt/live/${DOMAIN}/fullchain.pem ]; then
-  letsencrypt certonly \
-    --domain ${DOMAIN} \
-    --authenticator standalone \
-    ${SERVER} \
-    --email "${EMAIL}" --agree-tos
-fi
+  if [ ! -f /etc/letsencrypt/live/"${d}"/fullchain.pem ]; then
+    letsencrypt certonly \
+      --domain "${d}" \
+      --standalone \
+      --email "${EMAIL}" --agree-tos
+  fi
```

Comment:
This is a very good point! However, I'd say it's not a good idea, security-wise. Because if you share the same certificate between sercices and that certificate is compromised, then all the  services are compromised. Having different certs reduces the couplings between the served containers...

Oh, and I'm wondering on what the letsencrypt limitation is based. If it's based on the FQDN then you have plenty of room. If it's on the source IP then, indeed, we might hit the limit very quickly.